section_faiss_vector_index.go

```diff
@@ -609,20 +636,46 @@ type vecInfo struct {
 	docIDs *roaring.Bitmap
 }
 
-// todo: document the data structures involved in vector section.
 type vectorIndexOpaque struct {
 	init bool
 
+	lastNumVecs   int
+	lastNumFields int
+
+	// maps the field to the address of its vector section
 	fieldAddrs map[uint16]int
 
-	vecIDMap    map[int64]vecInfo
+	// maps the vecID to basic info involved around it such as
+	// the docIDs its present in and the vector itself
+	vecIDMap map[int64]vecInfo
+	// maps the field to information necessary for its vector
+	// index to be build.
 	vecFieldMap map[uint16]indexContent
 
 	tmp0 []byte
 }
 
-func (vo *vectorIndexOpaque) Reset() (err error) {
-	// cleanup stuff over here
+func (v *vectorIndexOpaque) allocateSpace() {
```

Comment:
basically, the reason why i had it as "allocateSpace" is because its not initializing or explicitly providing values for the fields of the struct. the function is allocating space for the datastructures involved without any assignment of values. 
the initialization with values is something which is something that happens in InitOpaque(). 
do you think changing it to just allocate()/realloc() is a good enough name?

Reply:
realloc() is good.

Reply:
i'll change this to realloc()